packaging/packaging_test.go

```diff
@@ -0,0 +1,782 @@
+				fileList, err = ioutil.ReadDir(reportDataPath)
+				if err != nil {
+					fmt.Println("Test files were not successfully created")
+					return err
+				}
+			} else {
+				fileList = nil
+			}
+
+			tmpDirMap := testDirConfig{
+				directory: reportPath,
+				files:     fileList,
+			}
+			if directory.dirName == "large" {
+				testDirs.large = tmpDirMap
+			} else if directory.dirName == "small" {
+				testDirs.small = tmpDirMap
+			} else if directory.dirName == "moving" {
+				testDirs.moving = tmpDirMap
+			} else if directory.dirName == "empty" {
+				testDirs.empty = tmpDirMap
+			} else if directory.dirName == "restricted" {
+				testDirs.restricted = tmpDirMap
+			} else if directory.dirName == "restrictedEmpty" {
+				testDirs.restrictedEmpty = tmpDirMap
+			} else if directory.dirName == "split" {
+				testDirs.split = tmpDirMap
+			} else if directory.dirName == "tar" {
+				testDirs.tar = tmpDirMap
+			}
+		}
+	}
+	return nil
+}
+
+func shutdown() {
+	fmt.Println("Tearing down for packaging tests")
+	os.RemoveAll(testingDir)
+}
+
+func TestMain(m *testing.M) {
+	err := setup()
+	if err != nil {
+		fmt.Println("Can not run tests because setup failed to create testing directories and files")
+		shutdown()
+	} else {
+		code := m.Run()
+		shutdown()
+		os.Exit(code)
+	}
+}
+
+func TestNeedSplit(t *testing.T) {
+	// create the needSplitTests
+	needSplitTests := []struct {
+		name     string
+		fileList []os.FileInfo
+		maxBytes int64
+		want     bool
+	}{
+		{
+			name:     "test split required",
+			fileList: testDirs.large.files,
+			maxBytes: 1 * 1024 * 1024,
+			want:     true,
+		},
+		{
+			name:     "test split not required",
+			fileList: testDirs.small.files,
+			maxBytes: 100 * 1024 * 1024,
+			want:     false,
+		},
+	}
+	for _, tt := range needSplitTests {
+		// using tt.name from the case to use it as the `t.Run` test name
+		t.Run(tt.name, func(t *testing.T) {
+			testPackager.maxBytes = tt.maxBytes
+			got := testPackager.needSplit(tt.fileList)
+			if tt.want != got {
+				t.Errorf("Outcome for test %s:\nReceived: %v\nExpected: %v", tt.name, got, tt.want)
+			}
+		})
+	}
+}
+
+func TestBuildLocalCSVFileList(t *testing.T) {
+	// create the buildLocalCSVFileList tests
+	buildLocalCSVFileListTests := []struct {
+		name     string
+		dirName  string
+		fileList []os.FileInfo
+	}{
+		{
+			name:     "test regular dir",
+			dirName:  filepath.Join(testDirs.large.directory, "data"),
+			fileList: testDirs.large.files,
+		},
+		{
+			name:     "test empty dir",
+			dirName:  filepath.Join(testDirs.empty.directory, "data"),
+			fileList: testDirs.empty.files,
+		},
+	}
+	for _, tt := range buildLocalCSVFileListTests {
+		// using tt.name from the case to use it as the `t.Run` test name
+		t.Run(tt.name, func(t *testing.T) {
+			got := testPackager.buildLocalCSVFileList(tt.fileList, tt.dirName)
+			want := make(map[int]string)
+			fileInfoList, _ := ioutil.ReadDir(tt.dirName)
```

Comment:
Even if we are not expecting an error with the test, we should still check it's return value.
```suggestion
			fileInfoList, err := ioutil.ReadDir(tt.dirName)
			if err != nil {
				t.Errorf("%s got unexpected error: %v", tt.name, err}
			}
```

Reply:
The ReadDir function for this test became obsolete so this is fixed.